tony-core/src/main/java/com/linkedin/tony/ReusablePort.java

```diff
@@ -107,23 +137,44 @@ private static int getAvailablePort() throws IOException {
    * SO_REUSEPORT.
    * @return the created port
    */
-  static ReusablePort create() throws IOException, InterruptedException {
+  static ReusablePort create() throws IOException {
     ReusablePort reusablePort;
     final int portBindingRetry = 5;
     for (int i = 0; i < portBindingRetry; i++) {
       try {
+        LOG.info("Port binding attempt " + (i + 1) + " ....");
         reusablePort = create(getAvailablePort());
         return reusablePort;
       } catch (BindException ex) {
-        LOG.info("port binding attempt " + (i + 1) + " failed.");
+        LOG.info("Port binding attempt " + (i + 1) + " failed.");
       }
     }
 
     throw new BindException("Unable to bind port after " + portBindingRetry + " attempt(s).");
   }
 
+  private static boolean waitTillPortReserved(int port) {
+    assert RESERVE_PORT_SCRIPT_PATH != null;
+    Path parentPath = RESERVE_PORT_SCRIPT_PATH.getParent();
+    assert parentPath != null;
+    Path fileToWait = Paths.get(parentPath.toString(), port + PORT_FILE_NAME_SUFFIX);
+
+    int checkCount = 0;
+    int maxCheckCount = 5;
+    while (!Files.exists(fileToWait) && (checkCount++) < maxCheckCount) {
```

Comment:
This is probably fine for now but won't inotify be better(given the port can be reserved during the sleep): https://docs.oracle.com/javase/tutorial/essential/io/notification.html

Reply:
i did look at inotify but given it only takes few secs to create the file, this is simple yet effective(no need to spend much time waiting).

Reply:
Ok but probably worth looking at this again.